libs/ml/src/ops/matrix_multiply.cpp

```diff
@@ -317,17 +318,17 @@ std::pair<OperationsCount, math::SizeVector> MatrixMultiply<T>::ChargeBackward(
 
   assert(this->batch_input_shapes_.size() == 2);
 
-  OperationsCount const n = this->batch_input_shapes_.front().at(0);
-  OperationsCount const m = this->batch_input_shapes_.back().at(0);
-  OperationsCount const p = 1;
+  OperationsCount const input_1_dim_1 = this->batch_input_shapes_.front().at(0);
+  OperationsCount const input_1_dim_2 = this->batch_input_shapes_.front().at(1);
+  OperationsCount const input_2_dim_1 = this->batch_input_shapes_.back().at(0);
+  OperationsCount const input_2_dim_2 = this->batch_input_shapes_.back().at(1);
+
+  OperationsCount op_cnt = 500;  // set up overhead
+  op_cnt += input_1_dim_1 * input_1_dim_2 * input_2_dim_1 * input_2_dim_2 *
+            fetch::ml::charge_estimation::ops::LOW_MULTIPLICATION_PER_ELEMENT;
 
-  OperationsCount const cost =
-      n * m * p * fetch::ml::charge_estimation::ops::LOW_MULTIPLICATION_PER_ELEMENT +
-      fetch::ml::charge_estimation::ops::LOW_ADDITION_PER_ELEMENT *
-          this->TotalElementsIn({this->batch_input_shapes_});
-  ;
   math::SizeVector output_shape = ComputeOutputShape(input_shapes);
-  return std::make_pair(cost * output_shape.back(), output_shape);
+  return std::make_pair(op_cnt * output_shape.back(), output_shape);
```

Comment:
I think that multiplying the setup cost by the batch size is probably incorrect.

Reply:
why? a call to backward potentially constructs new tensors. Seems like this would depend on batch size

Reply:
Oh okay, I thought it was an oversight and you only meant to multiply the tensor dimensions by the batch size.